pkg/extprom/http/instrument_client.go

```diff
@@ -0,0 +1,116 @@
+// Copyright (c) The Thanos Authors.
+// Licensed under the Apache License 2.0.
+
+package http
+
+import (
+	"net/http"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+)
+
+// ClientMetrics holds a collection of metrics that can be used to instrument a http client.
+// By setting this field in HTTPClientConfig, NewHTTPClient will create an instrumented client.
+type ClientMetrics struct {
+	inFlightGauge            prometheus.Gauge
+	requestTotalCount        *prometheus.CounterVec
+	dnsLatencyHistogram      *prometheus.HistogramVec
+	tlsLatencyHistogram      *prometheus.HistogramVec
+	requestDurationHistogram *prometheus.HistogramVec
+}
+
+// NewClientMetrics creates a new instance of ClientMetrics.
+// It will also register the metrics with the included register.
+// This ClientMetrics should be re-used for diff clients with the same purpose.
+// e.g. 1 ClientMetrics should be used for all the clients that talk to Alertmanager.
+func NewClientMetrics(reg prometheus.Registerer, namespace string) *ClientMetrics {
```

Comment:
I wonder if we can stick to per code metric naming. 

Essentially we can have `http_client_*` metrics  (no thanos prefix or anything) and use some `extprom.WrapRegistererWithLabel(reg, "client" "alertmanager") as reg parameter here.

Having stable name for all instances is powerful - allows greater metric stability and overall better dashboards/alerts/rules patterns reusability!


```suggestion
func NewClientMetrics(reg prometheus.Registerer) *ClientMetrics {
```

Reply:
aah this is a good suggestion 👌🏾